### ISSUE TITLE
CORS not working for multiple methods on same endpoint

### ISSUE THREAD

**Issue by oipat** [serverless/serverless#1960]
##### Serverless Framework Version: 
1.0.0-beta.2

##### Operating System: 
OSX 10.11.6

##### Expected Behavior:
OPTIONS request on endpoint returns Access-Control-Allow-Methods header with all configured methods enabled

##### Actual Behavior:
OPTIONS request on endpoint returns Access-Control-Allow-Methods header with only the method that is defined last in serverless.yml

Our serverless.yml config has the following function defined:

```yml
functions:
  blog:
    handler: blog/posts/handler.handler
    events:
      - http:
          path: posts
          method: get
          cors: true
      - http:
          path: posts
          method: post
          cors: true
      - http:
          path: posts/{id}
          method: put
          cors: true
      - http:
          path: posts/{id}
          method: delete
          cors: true
```

When requesting OPTIONS from the endpoint, only the DELETE method is listed as allowed:
curl -vv -X OPTIONS https://.....amazonaws.com/dev/posts/1234
->
..
< HTTP/1.1 200 OK
..
< Access-Control-Allow-Methods: OPTIONS,DELETE
..

##### TLDR:
CORS is not functioning properly when enabled for multiple methods on the same endpoint.

##### Discussion:
So it seems like only the latter definition for posts/{id} is valid for CORS. The actual PUT method does still work though. We also tried commenting out the definition for DELETE method and then the request for OPTIONS returned response with OPTIONS,PUT in the header.

Is there an error in our way of configuring or is it possibly a bug?

EDIT: fixed incorrect serverless version

**Comment by pmuens** [comment]
Thanks for reporting @oipat 

We're working on this in https://github.com/serverless/serverless/pull/1951 where we're refactoring the logical IDs for CloudFormation.

**Comment by flomotlik** [comment]
@oipat this is also definitely a bug as we're simply overriding the current cors config. As @pmuens already mentioned we're going to fix this with #1951 and a follow up that will merge the cors configs together properly.

At the moment the best thing you can do is only have CORS config on one endpoint and set the allowed methods there for all HTTP Methods. should work as a workaround for now.

**Comment by flomotlik** [comment]
This is fixed for the Method header with #1951 , but the Origins and header allow headers still fail here.
